modules/curl/8.7.1.bcr.1/patches/add_bazel_support.patch

```diff
@@ -0,0 +1,644 @@
++    "CURL_DISABLE_SOCKETPAIR",
++    "CURL_DISABLE_TELNET",
++    "CURL_DISABLE_TFTP",
++    "CURL_DISABLE_VERBOSE_STRINGS",
++    "CURL_WITH_MULTI_SSL",
++    "HAVE__FSEEKI64",
++    "HAVE_ADDRESS_FAMILY",
++    "HAVE_BROTLI",
++    "HAVE_CLOSESOCKET",
++    "HAVE_FSETXATTR_6",
++    "HAVE_GETPASS_R",
++    "HAVE_GLIBC_STRERROR_R",
++    "HAVE_GNUTLS_SRP",
++    "HAVE_GSSAPI",
++    "HAVE_GSSAPI_GSSAPI_GENERIC_H",
++    "HAVE_GSSAPI_GSSAPI_H",
++    "HAVE_GSSAPI_GSSAPI_KRB5_H",
++    "HAVE_GSSGNU",
++    "HAVE_IDN2_H",
++    "HAVE_IDNA_STRERROR",
++    "HAVE_IO_H",
++    "HAVE_IOCTLSOCKET",
++    "HAVE_IOCTLSOCKET_CAMEL",
++    "HAVE_IOCTLSOCKET_CAMEL_FIONBIO",
++    "HAVE_IOCTLSOCKET_FIONBIO",
++    "HAVE_LDAP_SSL_H",
++    "HAVE_LIBIDN2",
++    "HAVE_LIBSOCKET",
++    "HAVE_LIBSSH2",
++    "HAVE_LIBZ",  # TODO: Enable zlib support
++    "HAVE_LINUX_TCP_H",
++    "HAVE_OLD_GSSMIT",
++    "HAVE_OPENSSL_SRP",
++    "HAVE_QUICHE_CONN_SET_QLOG_FD",
++    "HAVE_SETSOCKOPT_SO_NONBLOCK",
++    "HAVE_SSL_SET0_WBIO",
++    "HAVE_STRCMPI",
++    "HAVE_STRICMP",
++    "HAVE_STROPTS_H",
++    "HAVE_SYS_FILIO_H",
++    "HAVE_SYS_SOCKIO_H",
++    "HAVE_SYS_UTIME_H",
++    "HAVE_TERMIO_H",
++    "HAVE_ZSTD",  # TODO: Enable zstd support
++    "in_addr_t",
++    "NEED_LBER_H",
++    "NEED_MALLOC_H",
++    "NEED_REENTRANT",
++    "PACKAGE",
++    "PACKAGE_BUGREPORT",
++    "PACKAGE_NAME",
++    "PACKAGE_STRING",
++    "PACKAGE_TARNAME",
++    "PACKAGE_VERSION",
++    "USE_ARES",
++    "USE_BEARSSL",
++    "USE_GNUTLS",
++    "USE_LIBPSL",
++    "USE_LIBSSH",
++    "USE_LIBSSH2",
++    "USE_MSH3",
++    "USE_NGHTTP2",
++    "USE_NGHTTP3",
++    "USE_NGTCP2",
++    "USE_OPENLDAP",
++    "USE_OPENSSL",
++    "USE_OPENSSL_QUIC",
++    "USE_QUICHE",
++    "USE_SCHANNEL",
++    "USE_SECTRANSP",
++    "USE_THREADS_WIN32",
++    "USE_TLS_SRP",
++    "USE_WEBSOCKETS",
++    "USE_WIN32_CRYPTO",
++    "USE_WIN32_IDN",
++    "USE_WIN32_LARGE_FILES",
++    "USE_WIN32_LDAP",
++    "USE_WINDOWS_SSPI",
++    "USE_WOLFSSL",
++    "VERSION",
++])
++
++_ENABLED_SETTINGS_MACOS = sets.make([
++    "HAVE_FSETXATTR_6",
++])
++
++_ENABLED_SETTINGS_LINUX = sets.make([
++    "HAVE_LINUX_TCP_H",
++])
++
++_ENABLED_SETTINGS_WINDOWS = sets.make([
++    "HAVE__FSEEKI64",
++    "HAVE_CLOSESOCKET",
++    "HAVE_IO_H",
++    "HAVE_IOCTLSOCKET",
++    "HAVE_IOCTLSOCKET_FIONBIO",
++    "HAVE_STRICMP",
++    "USE_SCHANNEL",
++    "USE_WIN32_IDN",
++    "USE_WINDOWS_SSPI",
++])
++
++_DISABLED_SETTINGS_MACOS = sets.make([
++    "HAVE_FSETXATTR_5",
++    "HAVE_GETHOSTBYNAME_R",
++    "HAVE_GETHOSTBYNAME_R_3",
++    "HAVE_GETHOSTBYNAME_R_5",
++    "HAVE_GETHOSTBYNAME_R_6",
++    "HAVE_MEMRCHR",
++    "HAVE_POLL_FINE",
++    "size_t",
++    "ssize_t",
++])
++
++_DISABLED_SETTINGS_LINUX = sets.make([
++    "size_t",
++    "ssize_t",
++])
++
++_DISABLED_SETTINGS_WINDOWS = sets.make([
++    "HAVE_ARC4RANDOM",
++    "HAVE_ARPA_INET_H",
++    "HAVE_ATOMIC",
++    "HAVE_BASENAME",
++    "HAVE_BOOL_T",
```

Comment:
I think the default curl config for windows does a better job with some of these defines 
https://github.com/curl/curl/blob/master/lib/config-win32.h#L113-L115

This patch breaks our project in compared to just using the default config.

```
ERROR: C:/users/zekew/_bazel_zekew/xrgzj2nk/external/ecsact_cli~/ecsact/cli/BUILD.bazel:28:10: Compiling ecsact/cli/ecsact_cli.cc failed: (Exit 2): cl.exe failed: 
error executing CppCompile command (from target @@ecsact_cli~//ecsact/cli:ecsact) C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.39.33519\bin\HostX64\x64\cl.exe ... (remaining 1 argument skipped)
C:\Users\zekew\_bazel_zekew\xrgzj2nk\execroot\_main\external\curl~\lib\curl_setup_once.h(242): error C2628: '<unnamed-enum-bool_false>' followed by 'bool' is illegal (did you forget a ';'?)
C:\Users\zekew\_bazel_zekew\xrgzj2nk\execroot\_main\external\curl~\lib\curl_setup_once.h(242): warning C4091: 'typedef ': ignored on left of '<unnamed-enum-bool_false>' when no variable is declared

```

Reply:
yea that's kinda tough. I think i fixed this case but it's not ideal how the feature defines are tied to these underlying defines that we don't really care to override at all.

Reply:
can you test with the new fix?

Reply:
sorry for the delay - I was away

Unfortunately I get a different error now:

```
C:\Users\zekew\_bazel_zekew\ejyagchf\execroot\_main\external\curl~\lib\curl_setup_once.h(257): error C2061: syntax error: identifier 'bit'                         
C:\Users\zekew\_bazel_zekew\ejyagchf\execroot\_main\external\curl~\lib\curl_setup_once.h(257): error C2059: syntax error: ';'
```

Reply:
ok how about now? I updated the PR to load the generated config, but then still allow the default configs to load too, which I think will always do a better job on these details that we don't really care about anyways

Reply:
same error. maybe some of the curl tests could be added to ensure the generated config works with the various platforms?

tested against `aa70252a81fb24bc4f01e4953cabfb5639a354ce`

Reply:
i dug a little further and I'm getting issues on ubuntu as well https://github.com/ecsact-dev/ecsact_cli/actions/runs/9026844616/job/24804753381?pr=93#step:4:85

using the fallback config like in #1917 seems to work though. I'd like to help where I can.

Reply:
@zaucy can you test with the newest version here? I am now loading both, but the variables we set from the curl_config.h could still be incorrect in some cases

Reply:
I think now that we're doing this we're good to merge this one and fixup new issues as they come up.

Reply:
I updated my repro, but it's still not working for us on Windows https://github.com/ecsact-dev/ecsact_cli/actions/runs/9374905263/job/25811997052?pr=93#step:4:655